packages/dev/core/src/assetContainer.ts

```diff
@@ -242,42 +360,29 @@ export class AssetContainer extends AbstractScene {
                     }
                 }
             }
-        };
 
-        this.meshes.forEach((o, idx) => {
-            if (localOptions.predicate && !localOptions.predicate(o)) {
-                return;
+            if (clone.parent === null) {
+                result.rootNodes.push(clone);
             }
+        };
 
-            if (!o.parent) {
-                const isInstance = o.getClassName() === "InstancedMesh";
-                let sourceMap: Mesh | undefined = undefined;
-                if (isInstance) {
-                    const oInstance = o as InstancedMesh;
-                    // find the right index for the source mesh
-                    const sourceMesh = oInstance.sourceMesh;
-                    const sourceMeshIndex = this.meshes.indexOf(sourceMesh);
-                    if (sourceMeshIndex !== -1 && instanceSourceMap[sourceMeshIndex]) {
-                        sourceMap = instanceSourceMap[sourceMeshIndex] as Mesh;
-                    }
-                }
-                const newOne = isInstance
-                    ? (o as InstancedMesh).instantiateHierarchy(
-                          null,
-                          {
-                              ...localOptions,
-                              newSourcedMesh: sourceMap,
-                          },
-                          onNewCreated
-                      )
-                    : o.instantiateHierarchy(null, localOptions, onNewCreated);
-
-                if (newOne) {
-                    if (instancesExist && newOne.getClassName() !== "InstancedMesh") {
-                        instanceSourceMap[idx] = newOne;
-                    }
-                    result.rootNodes.push(newOne);
+        sortedNodes.forEach((node) => {
+            if (node.getClassName() === "InstancedMesh") {
+                const instancedNode = node as InstancedMesh;
+                const sourceMesh = instancedNode.sourceMesh;
+                const replicatedSourceId = conversionMap[sourceMesh.uniqueId];
+                const replicatedSource = replicatedSourceId ? storeMap[replicatedSourceId] : sourceMesh;
+                const replicatedInstancedNode = replicatedSource.createInstance(instancedNode.name);
+                onNewCreated(instancedNode, replicatedInstancedNode);
+            } else {
+                // Mesh or TransformNode
+                const canInstance = !options?.doNotInstantiate && node.getClassName() === "Mesh" && (node as Mesh).getTotalVertices() > 0;
+                const replicatedNode = canInstance ? (node as Mesh).createInstance(node.name) : (node as TransformNode).clone(node.name, null, true);
+                if (!replicatedNode) {
+                    console.error("Could not clone or instantiate node on Asset Container", node.name);
+                    return;
                 }
+                onNewCreated(node as TransformNode, replicatedNode as TransformNode);

```

Comment:
nit: Maybe you could use `TransformNode` instead of `Node` in the new code? That would avoid the 3 "as TransformNode" casts